pkg/controller/deployment/util/deployment_util_test.go

```diff
@@ -585,52 +559,30 @@ func TestFindOldReplicaSets(t *testing.T) {
 			test:       "Get old ReplicaSets",
 			deployment: deployment,
 			rsList:     []*extensions.ReplicaSet{&newRS, &oldRS},
-			podList: &v1.PodList{
-				Items: []v1.Pod{
-					newPod,
-					oldPod,
-				},
-			},
-			expected: []*extensions.ReplicaSet{&oldRS},
+			expected:   []*extensions.ReplicaSet{&oldRS},
 		},
 		{
 			test:       "Get old ReplicaSets with no new ReplicaSet",
 			deployment: deployment,
 			rsList:     []*extensions.ReplicaSet{&oldRS},
-			podList: &v1.PodList{
-				Items: []v1.Pod{
-					oldPod,
-				},
-			},
-			expected: []*extensions.ReplicaSet{&oldRS},
+			expected:   []*extensions.ReplicaSet{&oldRS},
 		},
 		{
 			test:       "Get old ReplicaSets with two new ReplicaSets, only the oldest new ReplicaSet is seen as new ReplicaSet",
 			deployment: deployment,
 			rsList:     []*extensions.ReplicaSet{&oldRS, &newRS, &newRSDup},
-			podList: &v1.PodList{
-				Items: []v1.Pod{
-					newPod,
-					oldPod,
-				},
-			},
-			expected: []*extensions.ReplicaSet{&oldRS, &newRS},
+			expected:   []*extensions.ReplicaSet{&oldRS, &newRS},
 		},
 		{
 			test:       "Get empty old ReplicaSets",
 			deployment: deployment,
 			rsList:     []*extensions.ReplicaSet{&newRS},
-			podList: &v1.PodList{
-				Items: []v1.Pod{
-					newPod,
-				},
-			},
-			expected: []*extensions.ReplicaSet{},
+			expected:   nil,
 		},
 	}
 
 	for _, test := range tests {
-		old, _, err := FindOldReplicaSets(&test.deployment, test.rsList, test.podList)
+		_, old, err := FindOldReplicaSets(&test.deployment, test.rsList)
```

Comment:
We should test that the RSs are getting filtered correctly. This means that the inputs of this test should contain RSs with both rs.spec.replicas > 0 and rs.spec.replicas == 0 and then test that both outputs here conform to what we expect.